src/show_dialog.cpp

```diff
@@ -201,7 +201,7 @@ dialog_frame::dimension_measurements dialog_frame::layout(int x, int y, int w, i
 	}
 
 	y -= dim_.title.h;
-	w = std::max<int>(w,std::max<int>(int(dim_.title.w),int(buttons_width)));
+	w = std::max<int>(w,std::max<int>(dim_.title.w,int(buttons_width)));
```

Comment:
In this case, the compiler can decude the template argument. Both arguments are `int`s, for both `std::max()` calls.

I looked up the `int(buttons_width)` cast. It's called *function-style cast*, but it is semantically identical to C-style cast. See here: http://stackoverflow.com/a/32224